packages/core/src/node/file-uri.ts

```diff
@@ -36,8 +38,21 @@ export namespace FileUri {
         if (typeof uri === 'string') {
             return fsPath(new URI(uri));
         } else {
+            /*
+             * A uri for the root of a Windows drive, eg file:\\\c%3A, is converted to c:
+             * by the Uri class.  However file:\\\c%3A is unambiguously a uri to the root of
+             * the drive and c: is interpreted as the default directory for the c drive
+             * (by, for example, the readdir function in the fs-extra module).
+             * A backslash must be appended to the drive, eg c:\, to ensure the correct path.
+             */
             // tslint:disable-next-line:no-any
-            return (uri as any).codeUri.fsPath;
+            const fsPathFromVsCodeUri = (uri as any).codeUri.fsPath;
+            const isWindowsDriveRoot = windowsDriveRegex.exec(fsPathFromVsCodeUri);
```

Comment:
I think we should import `isWindows` from `@theia/core/lib/common/os` and run `exec` only if the OS is `win32`. What do you think?

Reply:
> I don't think that should ever happen

Yes. It should not happen. I would say, it is safe to add the `isWindows`.